functions/index.js

```diff
@@ -20,141 +23,26 @@ exports.sendPushNotification = functions.https.onRequest(async (req, res) => {
-      payload.apns.payload.aps.badge = 0;
-      payload.apns.payload.homeassistant = { 'command': 'clear_badge' };
-      updateRateLimits = false;
-    } else {
-      if(req.body.data) {
-        if (req.body.data.subtitle) {
-          payload.apns.payload.aps.alert.subtitle = req.body.data.subtitle;
-        }
-
-        if (req.body.data.push) {
-          for (var attrname in req.body.data.push) {
-            payload.apns.payload.aps[attrname] = req.body.data.push[attrname];
-          }
-        }
-
-        if(req.body.data.sound) {
-          payload.apns.payload.aps.sound = req.body.data.sound;
-        } else if(req.body.data.push && req.body.data.push.sound) {
-          payload.apns.payload.aps.sound = req.body.data.push.sound;
-        }
-
-        if (req.body.data.entity_id) {
-          payload.apns.payload.entity_id = req.body.data.entity_id;
-        }
-
-        if (req.body.data.action_data) {
-          payload.apns.payload.homeassistant = req.body.data.action_data;
-        }
-
-        if (req.body.data.attachment) {
-          payload.apns.payload.attachment = req.body.data.attachment;
-        }
-
-        if (req.body.data.url) {
-          payload.apns.payload.url = req.body.data.url;
-        }
-
-        if (req.body.data.shortcut) {
-          payload.apns.payload.shortcut = req.body.data.shortcut;
-        }
-
-        if (req.body.data.presentation_options) {
-          payload.apns.payload.presentation_options = req.body.data.presentation_options;
-        }
-      }
-
-      payload.apns.payload.aps.mutableContent = true;
-
-      if (req.body.message === 'delete_alert') {
-        updateRateLimits = false;
-        delete payload.notification.body;
-        delete payload.apns.payload.aps.alert.title;
-        delete payload.apns.payload.aps.alert.subtitle;
-        delete payload.apns.payload.aps.alert.body;
-        delete payload.apns.payload.aps.sound;
-      }
-    }
-  }
-
-  if(payload.apns.payload.aps.sound) {
-    if((typeof payload.apns.payload.aps.sound === "string") && (payload.apns.payload.aps.sound.toLowerCase() === "none")) {
-      delete payload.apns.payload.aps.sound;
-    } else if(typeof payload.apns.payload.aps.sound === "object") {
-      if(payload.apns.payload.aps.sound.volume) {
-        payload.apns.payload.aps.sound.volume = parseFloat(payload.apns.payload.aps.sound.volume);
-      }
-      if(payload.apns.payload.aps.sound.critical) {
-        payload.apns.payload.aps.sound.critical = parseInt(payload.apns.payload.aps.sound.critical);
-      }
-      if(payload.apns.payload.aps.sound.critical && payload.apns.payload.aps.sound.volume > 0) {
-        updateRateLimits = false;
-      }
-    }
-  }
-  if(payload.apns.payload.aps.badge) payload.apns.payload.aps.badge = Number(payload.apns.payload.aps.badge);
-  if(payload.apns.payload.aps.contentAvailable) {
-    payload.apns.headers['apns-push-type'] = 'background';
+    var response = ios.createPayload(req)
```

Comment:
```suggestion
    [updateRateLimits, response] = ios.createPayload(req)
```

But I would prefer if we use objects. It makes sure the values always map correctly.

In `android.js`: `return { updateRateLimits, payload }`

and here

```suggestion
    { updateRateLimits, payload } = ios.createPayload(req)
```